app/src/main/java/com/mobile/tandil/javabaseproject/activity/ReservationActivity.java

```diff
@@ -0,0 +1,50 @@
+package com.mobile.tandil.javabaseproject.activity;
+
+import android.os.Bundle;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import com.mobile.tandil.javabaseproject.database.ParkingDatabase;
+import com.mobile.tandil.javabaseproject.databinding.ActivityReservationBinding;
+import com.mobile.tandil.javabaseproject.entity.Reservation;
+import com.mobile.tandil.javabaseproject.listener.ListenerPickers;
+import com.mobile.tandil.javabaseproject.mvp.contract.ReservationContract;
+import com.mobile.tandil.javabaseproject.mvp.model.ReservationModel;
+import com.mobile.tandil.javabaseproject.mvp.presenter.ReservationPresenter;
+import com.mobile.tandil.javabaseproject.mvp.view.ReservationView;
+import java.util.Calendar;
+
+public class ReservationActivity extends AppCompatActivity implements ListenerPickers {
+
+    private ReservationContract.Presenter presenter;
+    private ActivityReservationBinding binding;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        binding = ActivityReservationBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        presenter = new ReservationPresenter(new ReservationView(this, binding), new ReservationModel(ParkingDatabase.getInstance()));
+        setListeners();
+    }
+
+    private void setListeners() {
+        binding.buttonStartingDayReservationActivity.setOnClickListener(view -> presenter.showPickers(this, true));
+        binding.buttonEndingDayReservationActivity.setOnClickListener(view -> presenter.showPickers(this, false));
+        binding.acceptButtonReservationActivity.setOnClickListener(view -> presenter.confirmReservation(
+                new Reservation(binding.textViewStartingDayReservationActivity.getText().toString(),
```

Comment:
The activity shouldn't create the reservation. It should pass the information. Please check who should have this responsability.